CODE-GUIDELINES.md

```diff
@@ -37,3 +37,35 @@ test('...', () => {
   vi.mocked(window.windowMethod).mockResolvedValue('a string');
 });
 ```
+
+### Do not use `try {} catch` in tests
+
+Let's have an async function such as
+
+```ts
+export function foo(): Promise<void> {
+  throw new Error('bar');
+}
+```
+
+Let's say we want to check that under certain condition the function throw an error (here always), we _could_ write the following code
+
+```ts
+// bad
+test('expect error to be properly catched and verified', async () => {
+  try {
+    await foo();
+  } catch(err: unknown) {
+    expect(err.message).toBe('bar')
+  }
+});
+```
+
+But it has a problem, if the code in the try block do not raise an error, we do not check the error message.
+Here is a version where we are sure the block rejected an error.
+
+```ts
+  await expect(async () => {
+    await foo();
+  }).rejects.toThrowError('Dummy Error');
+```
```

Comment:
from https://vitest.dev/api/expect#rejects it looks like it can be simplified 
(no async () => and you can use toThrow rather than toThrowError

```suggestion
  await expect(foo()).rejects.toThrow('Dummy Error');
```
```